[meta registration] fix _efficient_attention_forward for jagged inputs

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118657

Fixes the meta registration for the logsumexp output, whose shape should
be defined by the size of the offsets tensor when it exists.

https://github.com/pytorch/pytorch/blob/644f64f2d112b7c0b758b044821cf3972c0c17e9/aten/src/ATen/native/transformers/cuda/attention.cu#L1045

Differential Revision: [D53234217](https://our.internmc.facebook.com/intern/diff/D53234217)

davidberard98 committed:
[meta registration] fix _efficient_attention_forward for jagged inputs

Fixes the meta registration for the logsumexp output, whose shape should
be defined by the size of the offsets tensor when it exists.

https://github.com/pytorch/pytorch/blob/644f64f2d112b7c0b758b044821cf3972c0c17e9/aten/src/ATen/native/transformers/cuda/attention.cu#L1045

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53234217).

YuqingJ commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -8531,6 +8531,26 @@ def sample_inputs_efficient_attention_forward(op_info, device, dtype, requires_g
+            cu_seqlens_q=torch.tensor((0, 2, 4, 6, 8), dtype=torch.int32, device=device),
+            cu_seqlens_k=torch.tensor((0, 1, 4, 5, 6), dtype=torch.int32, device=device),
+            max_seqlen_q=2,
+            max_seqlen_k=2,
```
Not sure if it matters, the max_seqlen in your cu_seqlens_k is 3.

davidberard98 replied:
thanks for catching that

davidberard98 committed:
Update on "[meta registration] fix _efficient_attention_forward for jagged inputs"


Fixes the meta registration for the logsumexp output, whose shape should
be defined by the size of the offsets tensor when it exists.

https://github.com/pytorch/pytorch/blob/644f64f2d112b7c0b758b044821cf3972c0c17e9/aten/src/ATen/native/transformers/cuda/attention.cu#L1045

Differential Revision: [D53234217](https://our.internmc.facebook.com/intern/diff/D53234217)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53234217).

davidberard98 commented:
@pytorchbot merge

davidberard98 commented:
@pytorchbot merge

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53234217).
